core-rust/core-api-server/core-api-schema.yaml

```diff
@@ -1238,13 +1241,138 @@ components:
           $ref: "#/components/schemas/TransactionHeader"
         instructions:
           type: string
-          description: The decompiled transaction manifest instructions. Only returned if enabled in TransactionFormatOptions on your request.
+          description: The decompiled transaction manifest instructions. Only returned if enabled in `TransactionFormatOptions` on your request.
         blobs_hex:
           type: object
-          description: A map of the hex-encoded blob hash, to hex-encoded blob content. Only returned if enabled in TransactionFormatOptions on your request.
+          description: A map of the hex-encoded blob hash, to hex-encoded blob content. Only returned if enabled in `TransactionFormatOptions` on your request.
           additionalProperties:
             type: string
             description: The hex-encoded blob content
+        message:
+          description: The optional transaction message. Only returned if present and enabled in `TransactionFormatOptions` on your request.
+          $ref: "#/components/schemas/TransactionMessage"
+    TransactionMessageType:
+      type: string
+      enum:
+        - Plaintext
+        - Encrypted
+    TransactionMessage:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          $ref: "#/components/schemas/TransactionMessageType"
+      discriminator:
+        propertyName: type
+        mapping:
+          # NOTE: These need to match TransactionMessageType
+          Plaintext: '#/components/schemas/PlaintextTransactionMessage'
+          Encrypted: '#/components/schemas/EncryptedTransactionMessage'
+    PlaintextTransactionMessage:
+      allOf:
+        - $ref: "#/components/schemas/TransactionMessage"
+        - type: object
+          description: An unencrypted message.
+          required:
+            - mime_type
+            - content
+          properties:
+            mime_type:
+              type: string
+              description: The RFC 2046 MIME type of the `content`.
+            content:
+              $ref: "#/components/schemas/PlaintextMessageContent"
+    PlaintextMessageContentType:
+      type: string
+      enum:
+        - Unicode
```

Comment:
I actually mentioned `Unicode` since that's what Rust promises to support in its `String`.

However, I see that what really constraints us is "what can be decoded by SBOR from user's signed transaction bytes". And that is UTF-8 (https://github.com/radixdlt/radixdlt-scrypto/blob/285474de5ea0a3dbe8cac4e711b7c15a9059cd96/sbor/src/codec/string.rs#L43).

So I now used `String...` in the name, and adjusted the description to mention UTF-8, and I also removed the reference to not-actually-related `mime_type` 👍 

Please see if the new wording is proper English :p